applications/nrf_desktop/src/util/dfu_lock.h

```diff
@@ -9,14 +9,67 @@
 
 #include <zephyr/types.h>
 
+/**
+ * @defgroup dfu_lock DFU Lock API
+ * @brief DFU Lock API
+ *
+ * @{
+ */
+
+/** @brief DFU lock owner descriptor. */
 struct dfu_lock_owner {
+	/** Owner name. */
 	const char *name;
 
+	/** Notify the previous DFU owner that another DFU owner claimed the lock.
+	 *
+	 * This information can be used to perform the cleanup operations before
+	 * the subsequent DFU attempts of the previous DFU owner. Typically, the
+	 * previous DFU owner should reset its DFU progress and erase the DFU flash
```

Comment:
` and make sure that DFU flash memory is erased before writing new image content.`